File: src/cse/algorithms/Square.java
package cse.algorithms;

import java.util.ArrayList;
import java.util.Scanner;

public class Square {

    public static void main(String[] args) {
        Scanner scan=new Scanner(System.in);
        int cases=scan.nextInt();
        ArrayList<SandN> arrayList=new ArrayList<>();
        for (int i = 0; i < cases; i++) {
            int n=scan.nextInt();
            int s=scan.nextInt();
            arrayList.add(new SandN(n,s));
        }

        for (SandN sandN : arrayList) {
            System.out.println(sandN.numberOfSquares());
        }

    }
}

File: src/cse/algorithms/Square.kt
import java.util.*


object SquareCounting {
    @kotlin.jvm.JvmStatic
    fun main(args: Array<String>) {
        val scanner = Scanner(System.`in`)
        val cases = scanner.nextInt()
        val arrayList = ArrayList<SandN>()
        for (i in 0 until cases) {
            val n = scanner.nextInt()
            val s = scanner.nextInt()
            arrayList.add(SandN(n, s))
        }
        for (o in arrayList) {
            println(o.numberOfSquares())
        }
    }

    class SandN(private val n: Int, private val s: Int) {
        fun numberOfSquares(): Int {
            return if (n != 0) s / (n * n) else 0
        }
    }
}
